### PR TITLE
Add docs for creating new actions

### PR DESCRIPTION
**Pull request by panhania**
[Rendered](https://github.com/google/rrg/blob/docs/creating-actions/docs/creating-actions.md)

### CODE REVIEW

File: docs/creating-actions.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+an action that is internal to your deployment, use one of the field numbers from
+the reserved range between 1024 and 2048.
+
+Once the Protocol Buffers enum has the new field, add a new variant to the Rust
+`Action` enum defined in the [`rrg::request`] module. The compiler and the tests
+will guide you to towards updating the existing code to cover the new variant in
+all the required branches.
+
+As the last step, update the `dispatch` function in the [`rrg::request`] module
+and route the call to the `handle` function you defined. Remember to guard the
```
Comment (Tati1701) [inline docs/creating-actions.md:151]: I think this part could use an example, like you did above. I know it's probably less stable than the parts above, but... xD

**Review by Tati1701 — APPROVED** [review]
Hey, thanks for this one! Will help a lot when I start!!!

File: docs/creating-actions.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
+Creating actions
+================
+
+Creating new actions is easy but requires some boilerplate to be written first.
+You can take a look at commit 845c87b7c3373abacf41a17729ad95e1d6ab046a to see
```
[the remark this answers is not in the document]
Comment (panhania) [reply docs/creating-actions.md]: Already done! Before I created a pull request, I assumed that GitHub will hyperlink it automatically (like it does for commit references in comments and commit messages) but apparently I was wrong.

File: docs/creating-actions.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+an example of that. Follow the steps outlined in this guide for more details.
+
+### Define Protocol Buffers messages
+
+Each action should define its arguments and result as Protocol Buffer messages
+in its own package using the `proto3` syntax. All `.proto` definitions live in
+the `proto/rrg/action` subfolder. The file (and package) should be named as the
```
[the remark this answers is not in the document]
Comment (panhania) [reply docs/creating-actions.md:12]: Done.

File: docs/creating-actions.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+upstream, pick and use the first available field number. If you are developing
+an action that is internal to your deployment, use one of the field numbers from
+the reserved range between 1024 and 2048.
+
+Once the Protocol Buffers enum has the new field, add a new variant to the Rust
+`Action` enum defined in the [`rrg::request`] module. The compiler and the tests
+will guide you to towards updating the existing code to cover the new variant in
+all the required branches.
+
+As the last step, update the `dispatch` function in the [`rrg::request`] module
```
[the remark this answers is not in the document]
Comment (panhania) [reply docs/creating-actions.md:150]: Wrong module, should have been `rrg::action`. Fixed.

Unfortunately, there is no way to link to the symbol directly, as far as I know.

File: docs/creating-actions.md (hunk 1 above)
Comment (panhania) [reply docs/creating-actions.md:151]: The problem with the example here is that it too much depends on the surrounding code which is why I skipped it. I think with the link to the module fixed, it is actually pretty obvious what should be done.

**Review by Tati1701 — APPROVED** [review]

**Review by mbushkov — APPROVED** [review]

**Pull request merged**